### ISSUE TITLE
❓ [Question] "When the performance of an int8 model improves compared to an fp32 model after QAT"

### ISSUE THREAD

**Issue by JongSeok553** [pytorch/TensorRT#2062]
## ❓ Question

I have a question because there is something I do not understand during the QAT.

code ref: https://pytorch.org/TensorRT/_notebooks/vgg-qat.html#4

Phenomenon: The model with QAT applied and the simple TRT-converted model without QAT show higher accuracy than the fp32 model.
Data: 3-class dataset with approximately 210,000 images.
Model architecture: ResNet18.

Can the int8 converted TRT model perform better than the fp32 model?
![image](https://github.com/pytorch/TensorRT/assets/54762817/95276682-3525-4697-bd8e-29d6ea5cf7e7)

** Another question
## Environment

 - PyTorch Version (e.g., 1.0): v1.3.0
 - CPU Architecture: intel i9-10980
 - OS (e.g., Linux): ubuntu 20.04.3
 - How you installed PyTorch (`conda`, `pip`, `libtorch`, source): pip 
 - Build command you used (if compiling from source): 
 - Are you using local sources or building from archives:
 - Python version: 3.8
 - CUDA version: 11.6
 - GPU models and configuration: 
 - Any other relevant information:

**Comment by allihaider** [comment]
It seems that you are training all the models for the same number of epochs. Allowing the fp32 model to train for longer will likely improve its accuracy beyond that of a fully trained int8 quantized model. 

With the simple TRT-converted model that also stores its weights using fp32 precision, the results should be very close to that of a fully trained regular fp32 model, if not the same.

**Comment by JongSeok553** [comment]
thanks for the answer 
> It seems that you are training all the models for the same number of epochs. Allowing the fp32 model to train for longer will likely improve its accuracy beyond that of a fully trained int8 quantized model.
-> You mean the fp32 model will be higher than the int8 conversion model when trained enough?
     The fp32 model is a model trained for 200 epochs. 


> With the simple TRT-converted model that also stores its weights using fp32 precision, the results should be very close to that of a fully trained regular fp32 model, if not the same.
-> The simple trt model is also a model converted to int8, can it achieve similar accuracy to the fp32 model?

**Comment by JongSeok553** [comment]
Please check if the process of QAT I think is as follows.
1. Train fp32 model
2. qat_model = load(fp32_weights)
3. qat_model calibration(max_calibration)
4. qat_model train - 1-epoch
5. convert qat_model to TRT(int8)

I got the below result through the above process
fp32_trt_model is a case where fp32 model is calibrated and only trt conversion is performed without 1-epoch learning

The qat_trt model is a case where the fp32 model is calibrated, 1-epoch learned, and trt converted.
Both were converted to int8
![image](https://github.com/pytorch/TensorRT/assets/54762817/d1dbea5a-d9a9-445e-a2b7-5ffb45e0bbca)
